### PR TITLE
Fix survey validation

### PR DESCRIPTION
**Pull request by eliegaboriau**
#### :tophat: Description
*Please describe your pull request.*

This PR modify the survey validation flow : 
It now show the entire form when there is an error and scroll back to the first error.

#### :pushpin: Related Issues
*Link your PR to an issue*
- [Notion card](https://www.notion.so/opensourcepolitics/bd4bf860eab94f9daca7da803f09e3b2?v=b816eedbd04643008c0d78f580a0e6a0&p=5c6e574d88e146a1be1c588628a06adb&pm=c)

#### Testing
*Describe the best way to test or validate your PR.*

Example:
* Log in as admin
* Access Backoffice
* Go to survey settings
* Add mandatory text field and separator
* Go and anwser the survey
* Don't fill in a mandatory text field
* Submit and see you are redirected to the entire survey and to the first error
* fill in the mandatory text field but don't fill in a mandatory field other thant text
* Submit and see you are again redirected to the entire survey

#### Tasks
- [x] Add specs

### CODE REVIEW

**Review by paulinebessoles — DISMISSED** [review]
Thank you @eliegaboriau 👏

**Review by paulinebessoles — DISMISSED** [review]

File: app/cells/decidim/forms/step_navigation_cell.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+      def next_step_dom_id
+        "step-#{current_step_index + 1}"
+      end
+
+      def current_step_dom_id
+        "step-#{current_step_index}"
+      end
+
+      def errors
```
Comment (Quentinchampenois) [inline app/cells/decidim/forms/step_navigation_cell.rb:45]: We should implement this method in an extend

File: app/helpers/decidim/forms/application_helper.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        permited_models.include?(model_name)
+      end
+      alias show_public_participation? show_represent_user_group?
+
+      def permited_models
+        %(meeting)
+      end
+
+      def invalid?(responses)
+        bool = false
```
Comment (Quentinchampenois) [inline app/helpers/decidim/forms/application_helper.rb:20]:
```suggestion
                responses.map { |res| res.errors.any? }.any?
```

You can return an array of bool and just check if one of them is true

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
Thanks for the fix, product tests work as expected

I think we should limit overrides and use extends. Also can you add entries of added overrides to the OVERLOADS.md file please ?

**Review by paulinebessoles — DISMISSED** [review]

**Review by moustachu — APPROVED** [review]

**Review by Quentinchampenois — APPROVED** [review]

**Pull request merged**